cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/connectionStrategy/SSHConnectionStrategy.kt

```diff
@@ -0,0 +1,51 @@
+package com.cognifide.cogboard.widget.connectionStrategy
+
+import com.cognifide.cogboard.CogboardConstants.Props
+import com.cognifide.cogboard.ssh.SSHClient
+import io.vertx.core.json.Json
+import io.vertx.core.json.JsonObject
+
+class SSHConnectionStrategy(val config: JsonObject) : ConnectionStrategy() {
+
+    override fun getNumberOfLines(): Long? {
+        val logFilePath = config.getString(Props.PATH) ?: return null
+
+        return SSHClient(prepareConfig(config))
+                .executeAndClose("wc -l < $logFilePath")
+                ?.trim()
+                ?.toLongOrNull()
+    }
+
+    override fun getLogs(skipFirstLines: Long?): Collection<String> {
+        val logFilePath = config.getString(Props.PATH) ?: return emptyList()
+        val command = skipFirstLines?.let { "tail -n +${it + 1} $logFilePath" } ?: "cat $logFilePath"
```

Comment:
Question: Isn't this some potential problem when trying to load a huge file with `cat` command?